tianshou/data/buffer.py

```diff
@@ -123,17 +124,20 @@ class ReplayBuffer:
         This feature is not supported in Prioritized Replay Buffer currently.
     """
 
-    def __init__(self, size: int, stack_num: int = 1,
-                 ignore_obs_next: bool = False,
-                 save_only_last_obs: bool = False,
-                 sample_avail: bool = False) -> None:
+    def __init__(
+        self,
+        size: int,
+        stack_num: int = 1,
+        ignore_obs_next: bool = False,
+        save_only_last_obs: bool = False,
+        sample_avail: bool = False
+    ) -> None:
         super().__init__()
         self._maxsize = size
         self._indices = np.arange(size)
-        self._stack = None
         self.stack_num = stack_num
         self._avail = sample_avail and stack_num > 1
-        self._avail_index = []
+        self._avail_index: List[int] = []
```

Comment:
Are we really using variable typing in Tianshou ?

Reply:
This is the purpose of this PR. Remove this will cause mypy type-check error:
```
tianshou/data/buffer.py: note: In member "__init__" of class "ReplayBuffer":
tianshou/data/buffer.py:140: error: Need type annotation for '_avail_index' (hint: "_avail_index: List[<type>] = ...")  [var-annotated]
            self._avail_index = []
            ^
```